java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSubscriptionImpl.java

```diff
@@ -263,8 +254,13 @@ private void cancel() {
             return;
         }
 
-        resultTable.handleBarrageError(new RequestCancelledException("Barrage subscription cancelled by client"));
-        GrpcUtil.safelyCancel(observer, "Barrage subscription is cancelled", null);
+        final RequestCancelledException cancelledException =
+                new RequestCancelledException("Barrage subscription cancelled by client");
```

Comment:
Maybe we want to have a string argument to distinguish between "no longer live" and "cancelled by client" ?